cpp/src/binaryop/binaryop.cpp

```diff
@@ -411,6 +394,26 @@ bool is_same_scale_necessary(binary_operator op)
   return op != binary_operator::MUL && op != binary_operator::DIV;
 }
 
+template <typename Lhs, typename Rhs>
+void fixed_point_binary_operation_validation(binary_operator op,
+                                             Lhs lhs,
+                                             Rhs rhs,
+                                             thrust::optional<cudf::data_type> output_type = {})
+{
+  CUDF_EXPECTS(is_fixed_point(lhs), "Input must have fixed_point data_type.");
+  CUDF_EXPECTS(is_fixed_point(rhs), "Input must have fixed_point data_type.");
+  CUDF_EXPECTS(is_supported_fixed_point_binop(op), "Unsupported fixed_point binary operation");
+  CUDF_EXPECTS(lhs.id() == rhs.id(), "Data type mismatch");
```

Comment:
Is it illegal to add `decimal64 + decimal32 = decimal64`? Would the expectation be to cast the `decimal32` up to `decimal64` before doing the addition?

Reply:
We currently don't allow it. It is something we could support if users asked for it.